runtime/src/snapshot_bank_utils.rs

```diff
@@ -944,6 +949,45 @@ fn verify_slot_deltas_with_history(
     Ok(())
 }
 
+/// Verifies the bank's epoch stakes are valid after rebuilding from a snapshot
+fn verify_epoch_stakes(bank: &Bank) -> std::result::Result<(), VerifyEpochStakesError> {
+    _verify_epoch_stakes(
+        bank.epoch_stakes_map(),
+        bank.get_leader_schedule_epoch(bank.slot()),
+    )
+}
+
+/// Verifies the bank's epoch stakes are valid after rebuilding from a snapshot
+///
+/// This version of the function exists to facilitate testing.
+/// Normal callers should use `verify_epoch_stakes()`.
+fn _verify_epoch_stakes(
+    epoch_stakes_map: &HashMap<Epoch, EpochStakes>,
+    leader_schedule_epoch: Epoch,
+) -> std::result::Result<(), VerifyEpochStakesError> {
+    // Ensure epoch stakes from the snapshot does not contain entries for invalid epochs.
+    // Since epoch stakes are computed for the leader schedule epoch (`epoch + 1`), the snapshot's
+    // epoch stakes therefor can have entries for epochs at or below the leader schedule epoch.
+    if let Some(invalid_epoch) = epoch_stakes_map
+        .keys()
+        .find(|epoch| **epoch > leader_schedule_epoch)
+    {
+        return Err(VerifyEpochStakesError::EpochGreaterThanMax(
+            *invalid_epoch,
+            leader_schedule_epoch,
+        ));
+    }
+
+    // Ensure epoch stakes contains stakes for the leader schedule epoch
```

Comment:
Doesn't it also need an entry for the bank epoch to compute the leader schedule for the remainder of the epoch?